### PR TITLE
Fix links and simplify contributing.md

### PR DESCRIPTION
**Pull request by EliiseS**
## Checklist

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] Versions update reflected in all places (both __init__.py files, CHANGELOG, setup.py, setup.cfg)
- [x] Unit tests pass locally
- [x] Quickstart steps locally validated
- [x] Passes unit tests in CICD pipeline (green on Github pipeline)
- [x] Pypi RC version passes Edge CICD pipeline validation for cross-tool validation
- [x] Pull request includes test coverage for the included changes
- [x] Documentation is updated for the included changes

### General guidelines

- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.

## Description

After receiving feedback from the new contributors, I added missing information and simplified the contributing.md.

- Move docs file folders to kebab-case to match all the link in the docs and fix broken links when clicking on links in VSCode
  - in order for the links to work both in the wiki and VSCode the links must be formatted with just the file name (no .md), with the matching casing.
- Simplify contributing.md
- Deduplicate manual machine setup

### CODE REVIEW

File: docs/azure-setup.md [hunk 1]
```diff
@@ -31,25 +31,23 @@ iotedgedev iothub setup
...
-```
+```sh
 az ad sp create-for-rbac -n "iotedgedev01"
 ```
 
 > Note: Running `iotedgedev iothub setup` without any other parameters will save you time from looking up the required parameter values. The command will help you choose the parameters in an interactive way.
 
-Alternatively, you can deploy the IoT Hub **and** Container Registry with this **Deploy to Azure** template:
-
-[![Azure Deployment](https://azuredeploy.net/deploybutton.png)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure%2Fiotedgedev%2Fmaster%2Fassets%2Fdeploy%2FARMDeployment%2Fazuredeploy.json)
```
Comment (EliiseS) [inline docs/azure-setup.md:42]: Neither of these links work, but I only removed the image link for now: https://azuredeploy.net/deploybutton.png

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,121 +2,58 @@
...
-## Developing
+## Run Tests
 
-1. Run IoT Edge Dev Tool in editable mode
+You can choose one of these following commands to run the tests:
 
-    Run the following command from the root of the repo to see changes to iotedgedev commands as you change code.
+```sh
+# Run all tests with all python interpreters (fails for the ones not installed)
+# This is the command thats run in the pipeline for all python versions
```
Comment (marianan) [inline CONTRIBUTING.md:42]: >thats run [](http://example.com/codeflow?start=22&length=9)

nit: "that runs"

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,121 +2,58 @@
...
+# Run tests with tox in python version 3.9 (the python version installed in the devcontainer)
+tox -e py39
+# Run all tests with pytest for python version 3.9 (nicest output, fastest)
+make test
+```
 
-    ```
-    pip install -e .
-    ```
+In addition this this, you can also run and debug the tests individually using the VSCode test runner UI if the Python extension is installed (installed by default in the devcontainer).
```
Comment (marianan) [inline CONTRIBUTING.md:50]: >this this [](http://example.com/codeflow?start=12&length=9)

nit: remove

File: docs/azure-setup.md [hunk 4]
```diff
@@ -31,25 +31,23 @@ iotedgedev iothub setup
...
+Alternatively, you can deploy the IoT Hub **and** Container Registry with this **Deploy to Azure** template: [Azure Deployment](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure%2Fiotedgedev%2Fmaster%2Fassets%2Fdeploy%2FARMDeployment%2Fazuredeploy.json)
```
Comment (marianan) [inline docs/azure-setup.md:40]: >master [](http://example.com/codeflow?start=247&length=6)

should be main instead

**Review by marianan — APPROVED** [review]
:shipit:

File: docs/azure-setup.md (hunk 4 above)
Comment (EliiseS) [reply docs/azure-setup.md:40]: You're a wizard! I've also found the new location the the blue button, so now the image is back 🎉

**Pull request merged**